Cython/Utility/CommonStructures.c

```diff
@@ -7,90 +7,109 @@ static PyObject* __Pyx_FetchCommonTypeFromSpec(PyType_Spec *spec, PyObject *base
 
 /////////////// FetchCommonType ///////////////
 
-static PyObject *fetch_fake_module(void) {
-    PyObject *fake_module = PyImport_AddModule((char*) "_cython_" CYTHON_ABI);
-    if (!fake_module) return NULL;
-    Py_INCREF(fake_module);
-    return fake_module;
+static PyObject *__Pyx_FetchSharedCythonABIModule(void) {
+    PyObject *abi_module = PyImport_AddModule((char*) "_cython_" CYTHON_ABI);
+    if (!abi_module) return NULL;
+    Py_INCREF(abi_module);
+    return abi_module;
 }
 
-static void verify_cached_type(PyObject *cached_type,
+static int __Pyx_VerifyCachedType(PyObject *cached_type,
                                const char *name,
                                Py_ssize_t basicsize,
                                Py_ssize_t expected_basicsize) {
     if (!PyType_Check(cached_type)) {
         PyErr_Format(PyExc_TypeError,
             "Shared Cython type %.200s is not a type object", name);
+        return -1;
     }
     if (basicsize != expected_basicsize) {
         PyErr_Format(PyExc_TypeError,
             "Shared Cython type %.200s has the wrong size, try recompiling",
             name);
+        return -1;
     }
+    return 0;
 }
 
 static PyTypeObject* __Pyx_FetchCommonType(PyTypeObject* type) {
-    PyObject* fake_module;
+    PyObject* abi_module;
     PyTypeObject *cached_type = NULL;
 
-    fake_module = fetch_fake_module();
-    if (!fake_module) goto done;
-    cached_type = (PyTypeObject*) PyObject_GetAttrString(fake_module, type->tp_name);
+    abi_module = __Pyx_FetchSharedCythonABIModule();
+    if (!abi_module) goto bad;
+    cached_type = (PyTypeObject*) PyObject_GetAttrString(abi_module, type->tp_name);
     if (cached_type) {
-        verify_cached_type(
-            (PyObject *)cached_type,
-            type->tp_name,
-            cached_type->tp_basicsize,
-            type->tp_basicsize);
+        if (__Pyx_VerifyCachedType(
+              (PyObject *)cached_type,
+              type->tp_name,
+              cached_type->tp_basicsize,
+              type->tp_basicsize) < 0) {
+            goto bad;
+        }
         goto done;
     }
 
-    if (!PyErr_ExceptionMatches(PyExc_AttributeError)) goto done;
+    if (!PyErr_ExceptionMatches(PyExc_AttributeError)) goto bad;
     PyErr_Clear();
-    if (PyType_Ready(type) < 0) goto done;
-    if (PyObject_SetAttrString(fake_module, type->tp_name, (PyObject *)type) < 0)
-        goto done;
+    if (PyType_Ready(type) < 0) goto bad;
+    if (PyObject_SetAttrString(abi_module, type->tp_name, (PyObject *)type) < 0)
+        goto bad;
     Py_INCREF(type);
     cached_type = type;
 
 done:
+    Py_XDECREF(abi_module);
```

Comment:
`Py_DECREF()` is enough here, it's always non-NULL.

Reply:
Done!